### PR TITLE
feat: bank connect

### PR DESCRIPTION
**Pull request by dcocking**
### Type of change

- [ ] Chore
- [X] Feature
- [ ] Bug fix

### Linked issue
https://cybrid.atlassian.net/browse/CYB-847

### Why
We need to allow customers to connect a bank account.

### How
Added a bank-account-connect component that implements workflows, external_bank_accounts, and the Plaid-link SDK.

Screen Shot 2022-11-17 at 4 40 10 PM

Screen Shot 2022-11-17 at 4 41 09 PM

Screen Shot 2022-11-17 at 4 41 36 PM

### CODE REVIEW

File: library/src/app/components/bank-account-management/bank-account-connect/bank-account-connect.component.html [hunk 1]
```diff
@@ -0,0 +1,90 @@
+<ng-container *ngIf="configService.getConfig$() | async as config">
+  <mat-card-title>
+    <app-cybrid-logo
+      fill="{{ config.theme === 'LIGHT' ? '#0270e0' : 'white' }}"
```
Comment (brent-cybrid) [inline library/src/app/components/bank-account-management/bank-account-connect/bank-account-connect.component.html:4]: Shouldn't these colours be defined in css?

File: library/src/app/components/bank-account-management/bank-account-connect/bank-account-connect.component.spec.ts [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+  it('should handle errors when bootstrapping plaid', () => {
+    MockBankAccountService.getPlaidClient.and.returnValue(error$);
+
+    component.bootstrapPlaid('');
+
+    expect(MockErrorService.handleError).toHaveBeenCalled();
+    expect(MockEventService.handleEvent).toHaveBeenCalled();
+  });
+
+  it('should handle plaidOnSuccess()', () => {
```
Comment (brent-cybrid) [inline library/src/app/components/bank-account-management/bank-account-connect/bank-account-connect.component.spec.ts:159]: It's best practice to test one thing, i.e., success, invalid asset, or undefined asset, in each test as oppose to all in the same test.

Also, do we do any test to ensure we only got one account back?

File: src/demo/components/login/login.component.html [hunk 3]
```diff
@@ -90,6 +90,19 @@ <h1>Demo</h1>
             Customer not found
           </mat-error>
         </mat-form-field>
+
+        <mat-form-field appearance="outline">
```
Comment (brent-cybrid) [inline src/demo/components/login/login.component.html:94]: Do we need to set the bank guid if the customer guid has to be set? We'd need to validate the bank guid against the customer.

File: library/src/app/components/bank-account-management/bank-account-connect/bank-account-connect.component.html (hunk 1 above)
Comment (dcocking) [reply library/src/app/components/bank-account-management/bank-account-connect/bank-account-connect.component.html:4]: They are now css variable.

File: src/demo/components/login/login.component.html (hunk 3 above)
Comment (dcocking) [reply src/demo/components/login/login.component.html]: Now that the customer object includes bank ownership we no longer need to validate for this.

File: library/src/app/components/bank-account-management/bank-account-connect/bank-account-connect.component.spec.ts (hunk 2 above)
Comment (dcocking) [reply library/src/app/components/bank-account-management/bank-account-connect/bank-account-connect.component.spec.ts:161]: I've added a check for a single account.

File: library/src/app/components/bank-account-management/bank-account-connect/bank-account-connect.component.html [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+        <div class="connect-content">
+          <div>
+            <mat-icon color="warn">cancel</mat-icon>
+            <strong>{{ 'unexpectedError' | translate }}</strong>
+          </div>
+        </div>
+      </ng-container>
+      <ng-template #content>
+        <mat-stepper #stepper linear="true" animationDuration="0">
+          <!-- Connect Step -->
```
Comment (brent-cybrid) [inline library/src/app/components/bank-account-management/bank-account-connect/bank-account-connect.component.html:22]: We should use partials for these steps

File: library/src/app/components/bank-account-management/bank-account-connect/bank-account-connect.component.ts [hunk 5]
```diff
@@ -0,0 +1,279 @@
...
+    @Inject(DOCUMENT) public _document: Document,
+    public configService: ConfigService,
+    private errorService: ErrorService,
+    private eventService: EventService,
+    private bankAccountService: BankAccountService,
+    private workflowService: WorkflowsService,
+    private customersService: CustomersService,
+    private banksService: BanksService,
+    private router: RoutingService,
+    private _renderer2: Renderer2,
```
Comment (brent-cybrid) [inline library/src/app/components/bank-account-management/bank-account-connect/bank-account-connect.component.ts:89]: Odd name `renderer2`

File: library/src/app/components/bank-account-management/bank-account-connect/bank-account-connect.component.ts [hunk 6]
```diff
@@ -0,0 +1,279 @@
...
+    function isValidAsset(asset: string | null | undefined) {
+      return typeof asset == 'string';
+    }
+    function isOnlyAccount(accounts: any[]) {
+      return accounts.length == 1;
+    }
+
+    // const asset = 'USD';
+    const asset = metadata.accounts[0].iso_currency_code;
+    const account = metadata.accounts[0];
```
Comment (brent-cybrid) [inline library/src/app/components/bank-account-management/bank-account-connect/bank-account-connect.component.ts:216]: You'll want to guard this with the `isOnlyAccount` call so you don't get an error referencing an element in an empty list (or null list)

**Review by brent-cybrid — APPROVED** [review]

File: library/src/app/components/bank-account-management/bank-account-connect/bank-account-connect.component.ts (hunk 6 above)
Comment (dcocking) [reply library/src/app/components/bank-account-management/bank-account-connect/bank-account-connect.component.ts:216]: Good idea. I've moved them into the if statement.

File: library/src/app/components/bank-account-management/bank-account-connect/bank-account-connect.component.html (hunk 4 above)
Comment (dcocking) [reply library/src/app/components/bank-account-management/bank-account-connect/bank-account-connect.component.html:22]: This is a bit trickier that the identity-verification component. I've tried to pull the logic out but it doesn't result in it being easier to read, or really making it much shorter. I may make a general step partial that can function for any component going forward.

**Pull request merged**